From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Wed, 20 Jan 2021 11:55:10 +0100
Subject: window: Guard can_ping() against unmanaging windows

We remove pending pings when unmanaging a window, but currently
don't prevent new pings to be scheduled after that.

The previous commit fixed a code path where this did indeed happen,
but as the result of gnome-shell trying to attach a Clutter actor
to a non-existent window actor is pretty bad, also guard can_ping()
against being called for an unmanaging window.

Bug: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/2467
Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1676>
(cherry picked from commit 81f369480424fb5dd151f44c36104b3d237d789c)
Origin: upstream, 3.38.4, commit:75c3fc560eb7be9396dcf41bf7dd7c7d91b30417
---
 src/core/window.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/core/window.c b/src/core/window.c
index 1abad19..c87f8ee 100644
--- a/src/core/window.c
+++ b/src/core/window.c
@@ -8573,6 +8573,8 @@ meta_window_is_focusable (MetaWindow *window)
 gboolean
 meta_window_can_ping (MetaWindow *window)
 {
+  g_return_val_if_fail (!window->unmanaging, FALSE);
+
   return META_WINDOW_GET_CLASS (window)->can_ping (window);
 }
 
